core/templates/pages/collection-editor-page/navbar/collection-editor-navbar.directive.spec.ts

```diff
@@ -0,0 +1,544 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/**
+ * @fileoverview Unit tests for the Collection Editor Navbar directive.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { importAllAngularServices } from 'tests/unit-test-utils.ajs';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { UrlService } from 'services/contextual/url.service';
+import { Collection, CollectionBackendDict } from 'domain/collection/collection.model';
+import { CollectionEditorStateService } from '../services/collection-editor-state.service';
+import { CollectionRights } from 'domain/collection/collection-rights.model';
+import { UndoRedoService } from 'domain/editor/undo_redo/undo-redo.service';
+import { EventEmitter } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { CollectionRightsBackendApiService } from 'domain/collection/collection-rights-backend-api.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { CollectionValidationService } from 'domain/collection/collection-validation.service';
+// ^^^ This block is to be removed.
```

Comment:
Are we expecting these imports to be removed?

Reply:
This should be at ln: 23 (updated).